code/datums/components/uplink.dm

```diff
@@ -254,6 +248,12 @@
 
 	var/mob/user = arguments[2]
 	owner = "[user.key]"
+	if(owner)
+		LAZYINITLIST(GLOB.uplink_purchase_logs_by_key)
```

Comment:
i wish all this was on the mind, don't you? that's not really something you need to change, just me complaining into the void. i'd even say it'd be great to have on antag datum but sadly people who aren't antag need to have uplinks as well 🤷